src/NovelRT/CMakeLists.txt

```diff
@@ -111,41 +111,34 @@ target_include_directories(Engine
     $<INSTALL_INTERFACE:include>
 )
 
+target_compile_options(Engine
+  PRIVATE
+    $<$<CXX_COMPILER_ID:MSVC>:/W4>
+    $<$<CXX_COMPILER_ID:MSVC>:/WX>
+    $<$<CXX_COMPILER_ID:MSVC>:/wd4611>
+    $<$<CXX_COMPILER_ID:MSVC>:/MP>
+    $<$<CXX_COMPILER_ID:GNU>:-pedantic>
+    $<$<CXX_COMPILER_ID:GNU>:-pedantic-errors>
+    $<$<CXX_COMPILER_ID:GNU>:-Wall>
+    $<$<CXX_COMPILER_ID:GNU>:-Wextra>
+    $<$<CXX_COMPILER_ID:GNU>:-Werror>
+    $<$<CXX_COMPILER_ID:GNU>:-Wno-float-equal>
+    $<$<CXX_COMPILER_ID:GNU>:-Wno-padded>
+    $<$<CXX_COMPILER_ID:Clang>:-pedantic>
```

Comment:
Again, probably want duplicates for AppleClang.

Reply:
Done